src/components/CompoundInterest.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import TextField from '@material-ui/core/TextField';
+import {CompoundInterestChart, SavingsPlan} from './CompoundInterestChart';
+
+const initialState: SavingsPlan = {
+    monthlyInvestment: 300,
+    interestRate: 6.8,
+    durationInYears: 20,
+};
+
+type State = Readonly<SavingsPlan>;
+type InputChange = React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement | HTMLSelectElement>;
+
+export class CompoundInterest extends React.Component<{}, State> {
+    constructor(props: Readonly<{}>) {
+        super(props);
+        this.state = initialState;
+    }
+
+    public render() {
+        return (
+            <div>
+                <div>
+                    <TextField
+                        variant="outlined"
+                        defaultValue={initialState.monthlyInvestment}
```

Comment:
When using the value from the state directly we dont need the initialState variable.
```suggestion
                        value={this.state.monthlyInvestment}
```

Reply:
This changes the behaviour of the input fields. You can't insert negative numbers or numbers using scientific notation anymore. As I understand number-input fields, they only call the `onChange` handler, if the text field contains a valid number. As `-` is not a valid number, the change doesn't get propagated to the components state.

Reply:
Hmm, that sucks. It could be fixed with specifying two variables per field, one containing the last valid value and one containing the current value. But I'm okay with the current solution.